### PR TITLE
Fixes mobs that aren't able to speak being able to cast spoken spells

### LINKED ISSUE #51981: Wizards can cast spells requiring evocation while being unable to speak
round id 140971

Wizard mindswapped into a statue, something that is unable to speak, and was still able to cast fireball, a spell that needs to be spoken to be cast

### PR DESCRIPTION
**Pull request by ShizCalev**
:cl: ShizCalev
fix: Fixed player controlled statues being able to speak.
fix: Fixed nonhuman that aren't able to speak being able to cast spoken spells
/:cl:

fixes #51981

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/statue.dm [hunk 1]
```diff
@@ -109,6 +108,9 @@
 	if(!can_be_seen(get_turf(loc)))
 		..()
 
+/mob/living/simple_animals/hostile/statue/IsVocal() //we're a statue, of course we can't talk.
+	return
```
Comment (nemvar) [inline code/modules/mob/living/simple_animal/hostile/statue.dm:112]:
```suggestion
	return FALSE
```

File: code/modules/spells/spell.dm [hunk 2]
```diff
@@ -181,14 +181,15 @@ GLOBAL_LIST_INIT(spells, typesof(/obj/effect/proc_holder/spell)) //needed for th
 		to_chat(user, "<span class='warning'>[name] cannot be cast unless you are completely manifested in the material plane!</span>")
 		return FALSE
 
+	var/mob/living/L = user
+	if(istype(L) && (invocation_type == "whisper" || invocation_type == "shout") && !L.can_speak_vocal())
```
Comment (nemvar) [inline code/modules/spells/spell.dm:185]: Mind changing these invocation types to defines while you are at it?

**Comment by Tharcoonvagh** [comment]
Bye bye alien golem fuckwits

**Comment by ATH1909** [comment]
Does this mean we can let wizards mind transfer into statues again?

**Comment by ATH1909** [comment]
> Bye bye alien golem fuckwits

Are you sure that this affects alien golems/abductors?

**Review by nemvar — APPROVED** [review]

**Comment by ShizCalev** [comment]
@ATH1909 Yes.

**Comment by Wayland-Smithy** [comment]
I agree with @ATH1909 if a human can speak, but is in a vacuum does it still work? No sound was ever made so the universe at large should not recognize their magic? Alloy golems are still capable of using the `say` verb even if it is only detectable to Ayy telepathy users.

Something something tree falls down in the woods.

**Comment by Quietly12** [comment]
I disagree in the very large assumption being made here that spells necessarily need a verbal component to be heard by others to work. This assumption is obviously intended for balance, but as soon as you apply it, you realize how hard it breaks everything else.

Imagine we apply this, it means that spellcasters can be muted, gagged or otherwise prevented from speaking to prevent them from accessing magic. 

Mimes are punished very harshly under this set of rules (since they cannot play cult or heretic antagonists without a massive mood penalty), changelings can counter magical creatures effortlessly with mute sting, and any crew member can end a wizard's round by making a syringe gun with mute toxin.

**Comment by ShizCalev** [comment]
@Quietly12 They already can be. It just wasn't applied to non-human mobs. 

This just makes it also apply it to nonhuman mobs.

**Comment by Wayland-Smithy** [comment]
Correct me if I am wrong, but it seems current **human** wizards should already be ruined by losing their voice if there is an invoke component to the spell. Hence the issue in the first place. Personally I think this is fine since it does not punish creatures incapable of speaking in the first place.
```dm
// /code/modules/spells/spell.dm
if(ishuman(user))
	var/mob/living/carbon/human/H = user
	if((invocation_type == "whisper" || invocation_type == "shout") && !H.can_speak_vocal())
		to_chat(user, "<span class='warning'>You can't get the words out!</span>")
		return FALSE
```
As for the mime I would like to see them have to break their vow in order to take part in blood cult or heretic rituals, and the mood offset by their antag status.

**Comment by Quietly12** [comment]
> 
> 
> @Quietly12 They already can be. You're literally just describing the game as it already is, and how it has been for the last 10 years. It just wasn't checked for non-human mobs.

I was playing mime cult last week, it was fine without saying anything.

**Comment by optimumtact** [comment]
Nobody cares what you think, ship it

**Comment by Kaffe-work** [comment]
maybe mime should get a antag thing that removes their mime speak ban, much like how clowns do

**Pull request merged**